app/views/posts/index.html.erb

```diff
@@ -1 +1,69 @@
-<p>List of all Posts</p>
+<!-- Heading -->
+<h1 class="Head">Shows the details of posts for a given user</h1>
+
+<!-- User details section -->
+<section class='section-container'>
+  <div class='user'>
+    <!-- Profile photo -->
+    <div class='profile-div'>
+    <% if @user && @user.photo %>
+        <div class='profile-photo'><%= @user.photo %></div>
+    <% else %>
+        <div class='profile-photo'><%= image_tag('img-1.jpg', class: "profile-photo") %></div>
+    <% end %>
+    </div>
+    <!-- User name and post count -->
+    <div class='user_name'>
+      <% if @user %>
+        <h2><%= @user.name %></h2>
+        <div class='post_nums'>
+          <p class="post_count">Number of posts: <%= @user.posts_counter %></p>
+        </div>
+      <% else %>
+        <p>No user found.</p>
+      <% end %>
+    </div>
+    
+  <!-- Recent posts section -->
+  <section class="posts_container">
+    <% if @recent_posts.present? %>
```

Comment:
- [ ] I only can see no recent posts in your views, the reason because, you are colling recent posts a bit different that expected. What you need to do call it with @user.recent... that way your recent posts will be appeared.